src/Halcyon.WebApi/HAL/Json/JsonHALMediaTypeFormatter.cs

```diff
@@ -56,8 +56,25 @@ public class JsonHALMediaTypeFormatter : JsonMediaTypeFormatter {
                     value = halResponse.ToJObject(serializer);
                 }
             }
-            
+
             return base.WriteToStreamAsync(type, value, writeStream, content, transportContext);
         }
+
+        private bool TryGetHalResponse(Type type, object value, out HALResponse response) {
+            if (_converters != null) {
```

Comment:
We still need to check if it's an HALResponse without added converters. But I can make sure _converters are always initialized, thus removing the need for the null check